modules/data.remote/R/phenology_MODIS_extract.R

```diff
@@ -0,0 +1,115 @@
+##' Get MODIS phenology data by date and location  
+##' 
+##' @name phenology_MODIS_extract
+##' @title phenology_MODIS_extract
+##' @export
+##' @param settings PEcAn settings
+##' @param run_parallel optional method to download data parallely. Only works if more than 1 
+##' site is needed and there are >1 CPUs available.
+##' @param ncores number of cpus to use if run_parallel is set to TRUE. If you do not know the 
+##' number of CPU's available, enter NULL. 
+##' 
+##' The output file will be added to the folder defined in settings.
+##' 
+##' @examples
+##' \dontrun{
+##' settings <-  PEcAn.settings::read.settings("pecan.xml")
+##' test_modis_phenology <- PEcAn.data.remote::phenology_MODIS_extract(
+##'   settings = settings,
+##'   run_parallel = TRUE,
+##'   ncores = NULL)
+##' }
+##' @return NONE
+##' @author Qianyu Li
+##'
+
+phenology_MODIS_extract<- function(settings,run_parallel = TRUE,ncores = NULL){ 
+  #Get site id, lat, lon and site name for each site
+  site_info <- settings %>% 
+    purrr::map(~.x[['run']] ) %>% 
+    purrr::map('site')%>% 
+    purrr::map(function(site.list){
+      #conversion from string to number
+      site.list$lat <- as.numeric(site.list$lat)
+      site.list$lon <- as.numeric(site.list$lon)
+      list(site_id=site.list$id, lat=site.list$lat, lon=site.list$lon, site_name=site.list$name)
+      })%>% 
+    dplyr::bind_rows() %>% 
+    as.list()
+  #Set up the outdir to store leaf phenology data file           
+  outdir<-settings$model$leaf_phenology$outdir
+  if (!is.null(outdir)) {
```

Comment:
Nitpick: For cases like this where the`if` is really a condition check before proceeding down the happy path, it's often more readable to have the error case make the function exit early, then have the rest of the code below it.
```
if (is.null(outdir)) {
  PEcAn.logger::logger.error("No output directory found. Please set it up in pecan.xml under the model tag.")
}
start_date <- ...
```